### PR TITLE
add replica num limit per tablet

### PR DESCRIPTION
**Pull request by MorningLight5**
## Proposed changes

#7083

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [x] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

- [x] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/common/util/PropertyAnalyzer.java [hunk 1]
```diff
@@ -190,8 +190,9 @@ private static Short analyzeReplicationNum(Map<String, String> properties, Strin
                 throw new AnalysisException(e.getMessage());
             }
 
-            if (replicationNum <= 0) {
-                throw new AnalysisException("Replication num should larger than 0. (suggested 3)");
+            if (replicationNum < Config.min_replication_num_per_tablet || replicationNum > Config.max_replication_num_per_tablet) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/common/util/PropertyAnalyzer.java:193]: You may also need to modify the `analyzeReplicaAllocation()` method

**Review by morningman — DISMISSED** [review]
LGTM

**Comment by morningman** [comment]
hi @tianhui5 the FE unit test failed at DynamicPartitionTableTest.testSetDynamicPartitionReplicationNum.
You can see details here:
image

**Comment by tianhui5** [comment]
> hi @tianhui5 the FE unit test failed at DynamicPartitionTableTest.testSetDynamicPartitionReplicationNum. You can see details here: image

I've fix it.  Sorry I didn't see the button of log in as guest.

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**